staging/src/k8s.io/component-helpers/storage/volume/pv_helpers_test.go

```diff
@@ -193,58 +199,83 @@ func TestFindMatchVolumeWithNode(t *testing.T) {
 		claim           *v1.PersistentVolumeClaim
 		node            *v1.Node
 		excludedVolumes map[string]*v1.PersistentVolume
+		vacEnabled      bool
 	}{
 		"success-match": {
 			expectedMatch: "affinity-pv",
-			claim:         makeTestPersistentVolumeClaim("claim01", "100G", []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce}),
+			claim:         makeTestPersistentVolumeClaim("claim01", "100G", []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce}, nil),
 			node:          node1,
 		},
+		"success-match-empty-vac": {
+			vacEnabled:    true,
+			expectedMatch: "affinity-pv",
+			claim: makeTestPersistentVolumeClaim("claim01", "100G", []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce}, func(pvc *v1.PersistentVolumeClaim) {
+				pvc.Spec.VolumeAttributesClassName = ptr.To("")
+			}),
+			node: node1,
+		},
+		"success-match-vac": {
+			vacEnabled:    true,
+			expectedMatch: "affinity-pv2-vac",
+			claim: makeTestPersistentVolumeClaim("claim01", "100G", []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce}, func(pvc *v1.PersistentVolumeClaim) {
+				pvc.Spec.VolumeAttributesClassName = ptr.To("vac")
+			}),
+			node: node1,
+		},
 		"success-prebound": {
```

Comment:
From the latest kep updates, we should also validate VAC matches when PVC.spec.volumeName is set: https://github.com/kubernetes/enhancements/pull/4676/files#diff-60f3e23ac35e0c029253e5419717a00e20a4b309fa4fd364ae3b231585336c9eR476

@mattcary did we conclude on if it needs to match if PV.spec.claimRef is set? We don't do it for storageclass because we assume an admin knows what they're doing. But with VAC that may introduce some interesting behavior because we may try to modify the volume immediately after it binds.

Reply:
> From the latest kep updates, we should also validate VAC matches when PVC.spec.volumeName is set:

@msau42 I forget to change it. It's added in the latest commit.

Reply:
Yes, I think so. I don't see why we'd make an exception in that case when otherwise we only match if the VAC is consistent.

Reply:
ie the behavior during binding is unconditionally, the VAC must match.

Reply:
> ie the behavior during binding is unconditionally, the VAC must match.

@mattcary  fixed in the latest commit and add some unit tests.

Reply:
> ie the behavior during binding is unconditionally, the VAC must match.

@mattcary @msau42 updated and added an unit test for this case.